### PR TITLE
python3 compatibility

### PR DESCRIPTION
**Pull request by adnelson**
@narrativecraig I've been meaning to get this done for a little while

### CODE REVIEW

File: src/lsi/lsi.py [hunk 1]
```diff
@@ -58,19 +58,29 @@
...
+# python3. Oh well.
+if sys.version_info >= (3, 0):
+    from configparser import ConfigParser
+    from lsi.utils.hosts import (HostEntry, get_entries, filter_entries,
+                                 get_host)
+    from lsi.utils.term import (green, yellow, cyan, blue, get_input)
+    from lsi.utils.system import get_current_username
+    from lsi.utils.stream import stream_commands
+else:
+    from ConfigParser import ConfigParser
```
Comment (narrativecraig) [inline src/lsi/lsi.py:79]: Should ``from __future__ import print`` actually be in here?

File: src/lsi/lsi.py [hunk 2]
```diff
@@ -322,42 +332,36 @@ def _run_ssh(entries, username, idfile, no_prompt=False, command=None,
...
     return _PATHS[cmd]
 
 
 def _build_ssh_command(hostname, username, idfile, ssh_command, tunnel):
     """Uses hostname and other info to construct an SSH command."""
     command = [_get_path('ssh'),
                '-o', 'StrictHostKeyChecking=no',
-               '-o', 'ConnectTimeout=5',
-               '-o', 'UserKnownHostsFile={}'.format(_KNOWN_HOSTS_FILE)]
+               '-o', 'ConnectTimeout=5']
```
Comment (narrativecraig) [inline src/lsi/lsi.py:352]: As long as we're editing this file could you bump the timeout to 10 seconds?  I get banner timeouts semi-regularly on some of our slow/busy machines

**Review by narrativecraig — APPROVED** [review]
LGTM

**Comment by benjigoldberg** [comment]
@adnelson why dont we add haskal++ while we're at it

**Comment by adnelson** [comment]
@benjigoldberg all in good time

File: src/lsi/utils/system.py [hunk 3]
```diff
@@ -1,16 +1,16 @@
 # Copyright (c) 2015, Narrative Science
```
Comment (benjigoldberg) [inline src/lsi/utils/system.py:1]: not 2015

**Comment by benjigoldberg** [comment]
computer_wizard

**Comment by wilsoniya** [comment]
ashtray

File: src/lsi/utils/term.py [hunk 4]
```diff
@@ -75,7 +77,7 @@ def get_color_hash(string, _min=MIN_COLOR_BRIGHT, _max=MAX_COLOR_BRIGHT):
     """
     Hashes a string and returns a number between ``min`` and ``max``.
     """
-    hash_num = int(hashlib.sha1(string).hexdigest()[:6], 16)
+    hash_num = int(hashlib.sha1(string.encode('utf-8')).hexdigest()[:6], 16)
```
Comment (karmstrong-ns) [inline src/lsi/utils/term.py:80]: :feelsgood:

**Comment by karmstrong-ns** [comment]
img

File: src/lsi/lsi.py [hunk 5]
```diff
@@ -58,19 +58,29 @@
...
 
-from utils.hosts import HostEntry, get_entries, filter_entries, get_host
-from utils.term import (green, yellow, cyan, blue, get_input)
-from utils.system import get_current_username
-from utils.stream import stream_commands
-
+# The configparser module was renamed in python3. For the other
+# imports, they just seem to not work the same in python2 and
+# python3. Oh well.
+if sys.version_info >= (3, 0):
```
Comment (sfoulkes) [inline src/lsi/lsi.py:71]: Install the configparser module and you don't have to do this.

File: src/lsi/utils/system.py (hunk 3 above)
Comment (adnelson) [reply src/lsi/utils/system.py:1]: I'll let this one slide b/c yolo

File: src/lsi/lsi.py (hunk 5 above)
Comment (adnelson) [reply src/lsi/lsi.py:71]: meh

File: src/lsi/utils/system.py (hunk 3 above)
Comment (benjigoldberg) [reply src/lsi/utils/system.py:1]: watch out for the lsi internet pirates @adnelson ☠️

**Pull request merged**